sql/dd/impl/upgrade/server.cc

```diff
@@ -470,6 +470,67 @@ static void create_upgrade_file() {
 
 }  // namespace
 
+#ifdef WITH_WSREP
+
+bool pxc_fix_mysql_tables(THD *thd) {
+  const char section_start[] = "#! PXC_SECTION::START";
+  const char section_end[] = "#! PXC_SECTION::END";
+  int section_start_len = strlen(section_start);
+  int section_end_len = strlen(section_end);
+
+  const char **query_ptr;
+  bool in_pxc_section = false;
+
+  if (ignore_error_and_execute(thd, "USE mysql")) {
+    WSREP_ERROR("Could not run PXC upgrade. Could not change database to 'mysql'");
+    return true;
+  }
+
+  for (query_ptr = &mysql_fix_privilege_tables[0]; *query_ptr != NULL;
+       query_ptr++) {
+
+    if (!strncmp(*query_ptr, section_start, section_start_len)) {
+      in_pxc_section = true;
+    }
+    if (!strncmp(*query_ptr, section_end, section_end_len)) {
+      in_pxc_section = false;
+    }
+
+    if (in_pxc_section) {
+      if (ignore_error_and_execute(thd, *query_ptr)) return true;
```

Comment:
No additional error message in this case?

Reply:
Our code is a copy of fix_mysql_tables().  The errors are reported within the ignore_error_and_execute() function.